File: api/auth-service/src/main/java/com/carrental/authservice/model/dto/CreateUserDTO.java
package com.carrental.authservice.model.dto;

import com.carrental.authservice.annotation.PasswordMatches;
import com.carrental.authservice.annotation.ValidEmail;
import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
@PasswordMatches
public class CreateUserDTO {

    @NotNull
    @NotEmpty
    private String firstName;

    @NotNull
    @NotEmpty
    private String lastName;

    @NotNull
    @NotEmpty
    private String userName;

    @ValidEmail
    @NotNull
    @NotEmpty
    private String email;

    @NotNull
    @NotEmpty
    private String password;

    @NotNull
    @NotEmpty
    private String matchingPassword;
}

File: frontend-app/shared/model/CreateUserDTO.ts
export class CreateUserDTO {
    public firstName?: string;
    public lastName?: string;
    public email?: string;
    public phone?: string;
    public birthDate?: string;
    public password?: string;
    public matchingPassword?: string;

    constructor(
        firstName: string,
        lastName: string,
        email: string,
        phone: string,
        birthDate: string,
        password: string,
        matchingPassword: string
    ) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.phone = phone;
        this.birthDate = birthDate;
        this.password = password;
        this.matchingPassword = matchingPassword;
    }
}
